### DISCUSSION TITLE
jamulus can't see required libraries but ldd shows all dependencies met

### DISCUSSION THREAD

**Discussion by crantok** [jamulussoftware/jamulus#1930] [Technical Support & Help]
I've just installed jamulus for the first time. I installed the deb file and used apt-get to install dependencies. When I run jamulus it tells me that some dependencies are not found. This is on Ubuntu 16.04.

When I run ldd, it first prints the error messages from jamulus and then shows that all dependencies are met (except linux-vdso.so.1 ?)

Any idea what I should do? Thanks for any help.

```
ldd /usr/bin/jamulus
/usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libQt5Xml.so.5: version `Qt_5' not found (required by /usr/bin/jamulus)
/usr/bin/jamulus: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.27' not found (required by /usr/bin/jamulus)
/usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libQt5Gui.so.5: version `Qt_5' not found (required by /usr/bin/jamulus)
/usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.11' not found (required by /usr/bin/jamulus)
/usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.22' not found (required by /usr/bin/jamulus)
/usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libQt5Network.so.5: version `Qt_5' not found (required by /usr/bin/jamulus)
/usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5.9' not found (required by /usr/bin/jamulus)
/usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5' not found (required by /usr/bin/jamulus)
/usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5: version `Qt_5' not found (required by /usr/bin/jamulus)
	linux-vdso.so.1 =>  (0x00007ffcc5b0c000)
	libjack.so.0 => /usr/lib/x86_64-linux-gnu/libjack.so.0 (0x00007f1fddaba000)
	libQt5Widgets.so.5 => /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5 (0x00007f1fdd42d000)
	libQt5Gui.so.5 => /usr/lib/x86_64-linux-gnu/libQt5Gui.so.5 (0x00007f1fdcee5000)
	libQt5Network.so.5 => /usr/lib/x86_64-linux-gnu/libQt5Network.so.5 (0x00007f1fde205000)
	libQt5Xml.so.5 => /usr/lib/x86_64-linux-gnu/libQt5Xml.so.5 (0x00007f1fde1c9000)
	libQt5Core.so.5 => /usr/lib/x86_64-linux-gnu/libQt5Core.so.5 (0x00007f1fdca0f000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f1fdc7f2000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f1fdc470000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f1fdc167000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f1fdbf51000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f1fdbb87000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f1fde153000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f1fdb97f000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f1fdb77b000)
	libgobject-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007f1fdb528000)
	libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007f1fdb217000)
	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007f1fdaedd000)
	libpng12.so.0 => /lib/x86_64-linux-gnu/libpng12.so.0 (0x00007f1fdacb8000)
	libharfbuzz.so.0 => /usr/lib/x86_64-linux-gnu/libharfbuzz.so.0 (0x00007f1fdaa5a000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f1fda840000)
	libGL.so.1 => /usr/lib/x86_64-linux-gnu/mesa/libGL.so.1 (0x00007f1fda5cc000)
	libproxy.so.1 => /usr/lib/x86_64-linux-gnu/libproxy.so.1 (0x00007f1fda3ae000)
	libicui18n.so.55 => /usr/lib/x86_64-linux-gnu/libicui18n.so.55 (0x00007f1fd9f4c000)
	libicuuc.so.55 => /usr/lib/x86_64-linux-gnu/libicuuc.so.55 (0x00007f1fd9bb8000)
	libpcre16.so.3 => /usr/lib/x86_64-linux-gnu/libpcre16.so.3 (0x00007f1fd9952000)
	libffi.so.6 => /usr/lib/x86_64-linux-gnu/libffi.so.6 (0x00007f1fd974a000)
	libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007f1fd94da000)
	libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007f1fd92b8000)
	libfreetype.so.6 => /usr/lib/x86_64-linux-gnu/libfreetype.so.6 (0x00007f1fd900e000)
	libgraphite2.so.3 => /usr/lib/x86_64-linux-gnu/libgraphite2.so.3 (0x00007f1fd8de8000)
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007f1fd8bbf000)
	libxcb-dri3.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-dri3.so.0 (0x00007f1fd89bc000)
	libxcb-present.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-present.so.0 (0x00007f1fd87b9000)
	libxcb-sync.so.1 => /usr/lib/x86_64-linux-gnu/libxcb-sync.so.1 (0x00007f1fd85b2000)
	libxshmfence.so.1 => /usr/lib/x86_64-linux-gnu/libxshmfence.so.1 (0x00007f1fd83af000)
	libglapi.so.0 => /usr/lib/x86_64-linux-gnu/libglapi.so.0 (0x00007f1fd817e000)
	libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007f1fd7f6c000)
	libXdamage.so.1 => /usr/lib/x86_64-linux-gnu/libXdamage.so.1 (0x00007f1fd7d69000)
	libXfixes.so.3 => /usr/lib/x86_64-linux-gnu/libXfixes.so.3 (0x00007f1fd7b63000)
	libX11-xcb.so.1 => /usr/lib/x86_64-linux-gnu/libX11-xcb.so.1 (0x00007f1fd7961000)
	libxcb-glx.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-glx.so.0 (0x00007f1fd7748000)
	libxcb-dri2.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-dri2.so.0 (0x00007f1fd7543000)
	libXxf86vm.so.1 => /usr/lib/x86_64-linux-gnu/libXxf86vm.so.1 (0x00007f1fd733d000)
	libdrm.so.2 => /usr/lib/x86_64-linux-gnu/libdrm.so.2 (0x00007f1fd712b000)
	libicudata.so.55 => /usr/lib/x86_64-linux-gnu/libicudata.so.55 (0x00007f1fd5674000)
	libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007f1fd5470000)
	libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007f1fd526a000)
```

**Comment by rdica** [comment]
Ubuntu 16.04 is no longer supported by Canonical and is End Of Life. I believe the version of QT5 and supporting libs are too old based on the ldd output. You should upgrade to at least 18.04 LTS. If you cannot upgrade Ubuntu itself you might try compiling jamulus from src but I have no idea if that'll work, I don't have Ubuntu 16.04.

**Comment by crantok** [comment]
The ldd output shows that the qt5 deps are there. ldd can find them,
jamulus can't. I didn't install them manually so they were installed as
dependencies from the Ubuntu repos.

I take your point about end of life though.

**Reply by rdica** [reply]
/usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libQt5Xml.so.5: version `Qt_5' not found (required by /usr/bin/jamulus)
    /usr/bin/jamulus: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.27' not found (required by /usr/bin/jamulus)
    /usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libQt5Gui.so.5: version `Qt_5' not found (required by /usr/bin/jamulus)
    /usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.11' not found (required by /usr/bin/jamulus)
    /usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.22' not found (required by /usr/bin/jamulus)
    /usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libQt5Network.so.5: version `Qt_5' not found (required by /usr/bin/jamulus)
    /usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5.9' not found (required by /usr/bin/jamulus)
    /usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5' not found (required by /usr/bin/jamulus)
    /usr/bin/jamulus: /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5: version `Qt_5' not found (required by /usr/bin/jamulus)

To be clearer, the errors reported by ldd indicate that they cannot be found. The versions of libraries expected don't exist on the system.

**Comment by crantok** [comment]
As I said, the errors at the beginning of the output are not output from
ldd. Those are from the jamulus executable. All the lines following that
are ldd output and they show that the libraries are present.

Edit: Some versions of ldd load the executable in question, which is why
you should not run ldd on a suspect executable. ldd on Ubuntu is
evidently doing that.

**Comment by ghost** [comment]
If you installed the pre-compiled Jamulus package, the package was probably not compiled for Ubuntu 16.
Ubuntu 16 probably has different paths to the shared object (.so) libraries than later Ubuntu versions.
You must compile Jamulus from source. Hopefully the Jamulus source package can be compiled on older Ubuntu.
If you can't compile the Jamulus source package, the person who makes the source package should be notified.
I'm certain that you can compile the tarball... I can with CentOS 6 and Qt-5.6.1

**Comment by crantok** [comment]
Okay, I'll try that. Thanks.
